src/bin/pg_autoctl/keeper_config.c

```diff
@@ -579,6 +579,14 @@ keeper_config_set_setting(KeeperConfig *config,
 						  pg_is_not_running_is_ok))
 		{
 			config->pgSetup = pgSetup;
+			if (streq(path, "replication.password") &&
+				!keeper_config_validate_replication_password(config, value))
```

Comment:
The `keeper_config_set_setting` is only called from `cli_keeper_config_set` which is invoked when using the command line `pg_autoctl config set section.option value`. It is not used when loading values from the configuration file if it's been edited “at rest”, and it is not used at reload of the configuration file in the main server.